Support for Apple Push Notification Service

sam-io opened the pull request:
The APNS platform uses the Apple Push Notification service (APNS) to deliver
notifications from Home Assistant.

To use the APNS service you will need an apple developer account
and you will need to create an App to receive push notifications.
For more information see the apple developer documentation.
## Sample Configuration:

```
notify:
  platform: apns
  name: doorbell_app
  sandbox: true
  cert_file: doorbell_app.pem
```
## Configuration Variables:

**name:** The name of the app.
**sandbox:** If true notifications will be sent to the sandbox (test) notification service.
**cert_file:** The certificate to use to authenticate with the APNS service.
## Usage:

The APNS platform will register two services, notify/[app_name] and apns/[app_name].
### apns/app_name:

This service will register device id's with home assistant. In order to receive a notification a device must be registered. The app on the device can use this service to send its id during startup, the id will be stored in the [app_name]_apns.yaml.

See didRegisterForRemoteNotificationsWithDeviceToken in the apple developer documentation for more information.
### notify/app_name

This service will send messages to a registered device. The following parameters can be used:

**message:** The message to send

**target:** The desired state of the device, only devices that match the state will receive messages. To enable state tracking a registered device must have a device_tracking_id added to the [app_name]_apns.yaml file. If this id matches a device in known_devices.yaml its state will be tracked.

**data:**
**badge:** The number to display as the badge of the app.
**sound:** The name of a sound file in the app bundle or in the Library/Sounds folder.
**category:** Provide this key with a string value that represents the identifier property of the UIMutableUserNotificationCategory
**content_available:** Provide this key with a value of 1 to indicate that new content is available.

sam-io committed:
added push notification implementation

sam-io committed:
added push notification implementation

sam-io committed:
some lint changes

sam-io committed:
some lint changes

sam-io committed:
added docs

sam-io committed:
added docs

robbiet480 commented:
@sam-io What's the use case here? I assume you are writing an iOS app for Home Assistant? I've already started working on one and have gotten pretty far on it, but it is lacking in UI. It even has support for device tracking and super advanced push notifications with actions.

sam-io commented:
@robbiet480 it's just about notifications to an iOS app. I'm not writing a specific app for home assistant. I have written a simple app that will play a sound and notify me when someone rings my doorbell, I'm only distributing the app to people in my house so it's not published on the App Store or anything. I then wrote this extension to home assistant to allow me to send the app notifications.

robbiet480 commented:
@sam-io You may want to try out the HA iOS app, since it has push support and allows you to upload custom push notification sounds to use (as well as comes with [a ton of sounds by default](https://gist.github.com/robbiet480/a12be2ab6401f182952f0f95f6b229ff#file-list_of_sounds-txt)). 

Anyway, cool! Go ahead and fix up everything that @balloobbot caught and then I can take a look.

devdelay commented:
still rooting for a notification platform using iCloud messages... it can push a message with sound regardless of mute setting on the phone. It also uses what could be considered a more stable service than APNS. But for now I use @robbiet480 's iOS app and it's great!

sam-io committed:
Fixed comment formatting issues

sam-io commented:
Cool I will check out the app, didn't even know it existed, where can I get hold of it?

I have fixed up the issues but it looks like the CI build is failing because it can't load the module dependency, what have I missed?

kellerza commented:
Look at [requirements](https://home-assistant.io/developers/component_deps_and_reqs/), there is also a script to generate the `requirements_all.txt file`

A quick note on all the documentation, can you rather move it to the homeassistant.io project?

sam-io committed:
Added requirments

sam-io committed:
Update requirements_all.txt

robbiet480 commented:
@sam-io Sign up for beta testing the app [here](https://community.home-assistant.io/t/announcing-beta-testing-of-the-official-home-assistant-for-ios-app/411?u=robbiet480). Looking for contributors too, especially on the UI side!

sam-io committed:
Update apns.py

sam-io commented:
@kellerza thanks, that's what the problem was, I had the requirements defined at some point but must have deleted while moving some code around. Anyway, I can't re-run the script at the moment because I'm away from my computer for a week so I will have to pick it up, and the documentation when I get back. Thanks for the help so far.

sam-io committed:
re-generated requirments_all.txt

sam-io commented:
Ok, docs are done (see https://github.com/home-assistant/home-assistant.github.io/pull/905), anything else I need to do?

fabaff committed:
Minor Ecobee changes (#3131)

* Update configuration check, ordering, and constants

* Make API key optional

kellerza commented:
@robbiet480 would be the ideal person to comment here.

Quickly looking at it I would suggest removing the docs in line 2-70 and just put a link to the docs [example in [any](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/apcupsd.py) component]

Commit:
issue #3250

pvizeli committed:
Add voluptuous to ecobee (#3257)

fabaff committed:
Use constants and update ordering (#3261)

pvizeli committed:
Add support for complex template structures to data_template (#3255)

kellerza committed:
Improve yaml fault tolerance and handle check_config border cases (#3159)

fabaff committed:
Use voluptuous for nx584 alarm (#3231)

* Migrate to voluptuous

* Fix pylint issue

nkgilley committed:
fastdotcom from pypi (#3269)

fabaff committed:
Use constants and update ordering (#3268)

🐬

fabaff committed:
Use constants and update ordering (#3267)

🐬

lwis committed:
Add additional template for custom date formats (#3262)

I can live with a few visual line breaks 🐬

fabaff committed:
Use constants and update ordering (#3266)

rayrayndwiga committed:
Updated  braviatv's braviarc version to 0.3.5 (#3271)

fabaff committed:
Use voluptuous for Device Sun Light Trigger (#3105)

* Migrate to voluptuous

* Use default

Danielhiversen committed:
Merge pull request #3260 from home-assistant/issue_#3250

issue #3250

fabaff committed:
Point to master till archive is back (#3285)

fabaff committed:
Pi-Hole statistics sensor (#3158)

* Add Pi-Hole sensor

* Update docstrings and remove print()

* Use None for payload

turbokongen committed:
Added stuff for support range setting (#3189)

pvizeli committed:
cleanup Homematic code (#3291)

* cleanup old code

* cleanup round 2

* remove unwanted platforms

robbiet480 committed:
Update frontend

lwis committed:
Hotfix for #3100 (#3302)

snikch committed:
Fix TP-Link Archer C7 long passwords (#3225)

* Fix tplink C7 long passwords

Fixes an issue where passwords longer than 15 chars could not log in to Archer C7 routers.

* Truncate in correct place

* Add comment about TP-Link C7 pass truncation

* Fix lint error

* Truncate comment at 79 chars not 80

persandstrom committed:
modbus write registers service (#3252)

balloob committed:
Fix bloomsky platform discovery (#3303)

robbiet480 committed:
Remove dev tag

fabaff committed:
0.28 (#3288)

* Backend support for importing waypoints from owntracks as HA zones

* Added test for Owntracks waypoints import

* Backend support for importing waypoints from owntracks as HA zones

* Added test for Owntracks waypoints import

* Removed redundant assignment to CONF_WAYPOINT_IMPORT_USER

* Fixed zone test break and code style issues

* Fixed style issues

* Fixed variable scope issues for entities

* Fixed E302

* Do not install pip packages in tests

* EventBus: return function to unlisten

* Convert automation to entities with services

* Refactored zone creation based on code review feedback, enhanced configuration

* Added unit test to enhance waypoint_whitelist coverage

* Fix JSON encoder issue in recorder

* Fix tests docstring

* * Improved zone naming in waypoint import
* Added more test coverage for owntracks and zone

* Back to 0.28.0.dev0

* Code review feedback from @pavoni

* Added bitfield of features for flux_led since we are supporting effects

* Host should be optional for apcupsd component (#3072)

* Use voluptuous for file (#3049)

* Zwave climate Bugfix: if some setpoints have different units, we should fetch the o… (#3078)

* Bugfix: if some setpoints have different units, we should fetch the one that are active.

* Move order of population for first time detection

* Default to config if None unit_of_measurement

* unit fix (#3083)

* humidity slider (#3088)

* If device was off target temp was null. Default to Heating setpoint (#3091)

* Fix linting

* Upgrade pyuserinput to 0.1.11 (#3068)

* Upgrade pyowm to 2.4.0 (#3067)

* improve isfile validation check (#3101)

* Refactor notification titles to allow for them to be None, this also includes a change in Telegram to only include the title if it's present, and to use a Markdown parse mode for messages (#3100)

* Fix broken test

* rfxtrx sensor clean up

* Bitcoin sensor use warning instead of error (#3103)

* Use voluptuous for HDMI CEC & CONF_DEVICES constants (#3107)

* Update voluptuous for nest (#3109)

* Update configuration check
* Extend platform

* Fix for BLE device tracker (#3019)

* Bug fix tracked devices
* Added scan_duration configuration parameter

* fix homematic climate implementation (#3114)

* Allow 'None' MAC to be loaded from known_devices (#3102)

* Use voluptuous for xmpp (#3127)

* Use voluptuous for twitter (#3126)

* Use voluptuous for Fritzbox and DDWRT (#3122)

* Use Voluptuous for BT Home Hub (#3121)

* Use voluptuous for syslog (#3120)

* Use voluptuous for Aruba (#3119)

* Use constants, update configuration check, and ordering (Pilight) (#3118)

* Use contants, update configuration check, and ordering

* Fix pylint issue

* Migrate to voluptuous (#3113)

* Fix typo (#3108)

* Migrate to voluptuous (#3106)

* Update voluptuous (#3104)

* Climate and cover bugfix (#3097)

* Avoid None comparison for zwave cover.

* Just rely on unit from config for unit_of_measurement

* Explicit return None

* Mqtt (#11)

* Explicit return None

* Missing service and wrong service name defined

* Mqtt state was inverted, and never triggering

* Migrate to voluptuous (#3096)

* Migrate to voluptuous (#3084)

* Fixed Homematic cover (#3116)

* Migrate to voluptuous (#3069)

🐬

* Migrate to voluptuous (#3066)

🐬

* snapcast update (#3012)

* snapcast update

* snapcast update

* validate config

* use conf constants

* orvibo updates (#3006)

🐬

* Update frontend

* move units to temperature for climate zwave. wrong state was sent to mqtt cove

* Use voluptuous for instapush (#3132)

* Use voluptuous for Octoprint (#3111)

* Migrate to voluptuous

* Fix pylint issues

* Add missing docstrings (fix PEP257 issues) (#3098)

* Add missing docstrings (fix PEP257 issues)

* Finish sentence

* Updated braviatv's braviarc version to 0.3.4 (#2997)

* Updated braviarc version to 0.3.4

* Updated braviarc version to requirements_all.txt

* Use voluptuous for Acer projector switch (#3077)

🐬

* Use voluptuous for twilio (#3134)

* Use voluptuous for webostv (#3135)

* Use voluptuous for Command line platforms (#2968)

* Migrate to voluptuous

* Fix pylint issues

* Remove FIXME

* Split setup test

* Test with bootstrap

* Remove lon and lat

* Fix pylint issues

* Add coinmarketcap sensor (#3064)

* Migrate to voluptuous (#3142)

🐬

* Back out insteon hub and fan changes (#3062)

* Move details to docs (#3146)

* Update frontend

* Use constants (#3148)

* Update ordering (#3149)

* Migrate to voluptuous (#3092)

* Display the error instead of the traceback (notify.slack) (#3079)

* Display the error instead of the traceback

* Remove name for check

* Automatic ODB device tracker & device tracker attributes (#3035)

* Migrate to voluptuous (#3173)

* Add voluptuous for tomato and SNMP (#3172)

* Improve voluptuous and login errors for Asus device tracker (#3170)

* Add exclude option to nmap device tracker (#2983)

* Add exclude option to nmap device tracker

Adds an optional exclude paramater to nmap device tracker.
Devices specified in the exclude list will never be scanned
by nmap. This can help to reduce log spam.

ex:
```
device_tracker:
  - platform: nmap_tracker
    hosts: 10.0.0.1/24
    home_interval: 1
    interval_seconds: 12
    consider_home: 120
    track_new_devices: yes
    exclude:
      - 10.0.0.2
      - 10.0.0.1
```

* Handle optional exclude

* Style fixed

* Added Xbox Live component (#3013)

* Added Xbox Live component

* Added Xbox Live sensor to coveralls

* Added init success checks

* Added entity id

* Adding link_names to post.message call (#3167)

If you do not turn link_names on, Slack will not highlight @channel and @username messages.

* Allow https (fixes #3150) (#3155)

* Use constants (#3156)

* Bugfix: ctach Runtime errors (#3153)

"RuntimeError: Disable scan failed" has been seen in a live installation

* Migrate to voluptuous (#3166)

🐬

* Migrate to voluptuous (#3164)

🐬

* Migrate to voluptuous (#3163)

🐬

* Migrate to voluptuous (#3162)

🐬 and 🍪 for fixing quotes!

* Exclude www_static from pydocstyle linting (#3175)

🐬

* Migrate to voluptuous (#3174)

* Migrate to voluptuous (#3171)

* Use voluptuous for mFi switch (#3168)

* Migrate to voluptuous

* Take change configuration into account

* Migrate to voluptuous (#3144)

🐬

* Add the occupancy sensor_class (#3176)

Such a complicated PR

* Update frontend

* Use voluptuous for Unifi, Ubus (#3125)

* Using alert with Hue maintains prior state (#3147)

* When using flash with hue, dont change the on/off state of the light so that it will naturally return to its previous state once flash is complete

* ATTR_FLASH not ATTR_EFFECT

* MQTT fan platform (#3095)

* Add fan.mqtt, allow brightness to be passed and mapped to a fan speed for compatibility with emulated_hue

* Pylint/Flake8 fixes

* Remove brightness

* Add more features, like custom oscillation/speed payloads and setting the speed list

* Flake8 fixes

* flake8/pylint fixes

* Use constants

* block fan.mqtt from coverage

* Fix oscillating comment

* Add Sphinx API doc generation (#3029)

* add's sphinx project to docs/ dir
* include core/helpers autodocs for API reference

* Allow reloading automation without restarting HA (#3002)

* Migrate to voluptuous (#3182)

🐬

* Migrate to voluptuous (#3179)

🐬

* Added scale and offset to the Temper component (#2853)

🐬

* Use voluptuous for BT and Owntracks device trackers (#3187)

🐬

* Correct binary_sensor.ecobee docs URL

* Use voluptuous for Hikvisioncam switch (#3184)

* Migrate to voluptuous

* Use vol.Optional

* Use voluptuous for Edimax (#3178)

🐬

* Use voluptuous for Bravia TV (#3165)

🐬

* Added support to 'effect: random' to Osram Lightify lights (#3192)

* Added support to 'effect: random' to Osram Lightify lights

* removed extra line not required

* Use voluptuous for message_bird, sendgrid (#3136)

* Try out the RTD theme

* Doc updates

* Update voluptuous for existing notify platforms (#3133)

* Update voluptuous for exists notify platforms

* fix constants

* Simple trend sensor. (#3073)

* First cut of trend sensor.

* Tidy.

* Migrate to voluptuous (#3193)

* Migrate to voluptuous (#3194)

🐬

* Migrate to voluptuous (#3197)

* Migrate to voluptuous (#3198)

🐬

* Use extend of PLATFORM_SCHEMA (#3199)

* Migrate to voluptuous (#3202)

🐬

* Updated to use the occupancy sensor_class (#3204)

🐬

* Migrate to voluptuous (#3206)

* Migrate to voluptuous (#3207)

* Migrate to voluptuous (#3208)

🐬

* Migrate to voluptuous (#3209)

🐬

* Migrate to voluptuous (#3214)

* Use voluptuous for SqueezeBox (#3212)

* Migrate to voluptuous

* Remove name

* Migrate to voluptuous and upgrade uber_rides to 0.2.5 (#3181)

* Migrate to voluptuous (#3200)

🐬

* Use Voluptuous for Luci and Netgear device trackers (#3123)

* Use Voluptuous for Luci and NEtgear device trackers

* str_schema shortcut

* Undo str_schema

* change update handling with variable for breack CCU2 (#3215)

* Update ordering (#3216)

* Docs update

* Flake8/pylint

* Add new docs requirements

* Update email validation (#3228)

🐬

* Fix email validation (fixes #3138) (#3227)

* Upgrade slacker to 0.9.25 (#3224)

* Upgrade psutil to 4.3.1 (#3223)

* Upgrade gps3 to 0.33.3 (#3222)

* Upgrade Werkzeug to 0.11.11 (#3220)

* Upgrade sendgrid to 3.4.0 (#3226)

* Bluetooth: keep looking for new devices (#3201)

* keep looking for new devices

* Update bluetooth_tracker.py

* change default value for tracking new devices

* remove commented code

* dlink switch added device state attributes and support for legacy firmware (#3211)

* Use voluptuous for free mobile (#3236)

* Use voluptuous for nma (#3241)

* Improve 1-Wire device family detection and error checking. Use volupt… (#3233)

* Improve 1-Wire device family detection and error checking. Use voluptuous

* Fix detection of gpio connected devices

* Replace rollershutter and garage door with cover, add fan (#3242)

* Use voluptuous for Alarm.com (#3229)

* Use voluptuous for gntp (#3237)

* Use voluptuous for pushbullet, pushetta and pushover (#3240)

* Migrate to voluptuous (#3230)

🐬

* Fix mFi sensors in uninitialized state (#3246)

If mFi sensors are identified but not fully assigned they can
have no tag value, and mficlient throws a ValueError to signal this.
This patch handles that case by considering such devices to always
be STATE_OFF.

* Use voluptuous for PulseAudio Loopback (#3160)

* Migrate to voluptuous

* Fix conf var

* Use voluptuous for Verisure (#3169)

* Migrate to voluptuous

* Update type and add missing config variable

* thread safe modbus (#3188)

*  Upgraded fitbit to version 0.2.3 which fixed oauthlib.oauth2.rfc6749.errors.TokenExpiredError: (token_expired) (#3244)

* update ffmpeg version to 0.10 add get image to camera (#3235)

* Migrate to voluptuous (#3234)

* fix bugfix with unique_id (#3217)

* Zwave climate fix and wink cover. (#3205)

* Fixes setpoint get was done outside loop

* zxt_120

* Wink not migrated to cover

* Clarifying debug

* too long line

* Only add 1 device entity

* Owntracks voluptuous fix (#3191)

* Zwave set temperature fix (#3221)

* If device was off set target temp would not work.

* Changed to use a workaround just for Horstmann HRT4-ZW Zwave Thermostat

* Wrong Horseman id

* style changes

* Change PR to suggestion on gitter (#3243)

* Reload groups (#3203)

* Allow reloading groups without restart

* Test to make sure automation listeners are removed.

* Remove unused imports for group tests

* Simplify group config validation

* Add prepare_reload function to entity component

* Migrate group to use entity_component.prepare_reload

* Migrate automation to use entity_component.prepare_reload

* Clean up group.get_entity_ids

* Use cv.boolean for group config validation

* fix remove listener (#3196)

* Add linux battery sensor (#3238)

* protect service data for changes in calls (#3249)

* protect service data for changes in calls

* change handling

* move MappingProxyType to service call

* Fix issue #3250 (#3253)

* Minor Ecobee changes (#3131)

* Update configuration check, ordering, and constants

* Make API key optional

* issue #3250

* Add voluptuous to ecobee (#3257)

* Use constants and update ordering (#3261)

* Add support for complex template structures to data_template (#3255)

* Improve yaml fault tolerance and handle check_config border cases (#3159)

* Use voluptuous for nx584 alarm (#3231)

* Migrate to voluptuous

* Fix pylint issue

* fastdotcom from pypi (#3269)

* Use constants and update ordering (#3268)

🐬

* Use constants and update ordering (#3267)

🐬

* Add additional template for custom date formats (#3262)

I can live with a few visual line breaks 🐬

* Use constants and update ordering (#3266)

* Updated  braviatv's braviarc version to 0.3.5 (#3271)

* Use voluptuous for Device Sun Light Trigger (#3105)

* Migrate to voluptuous

* Use default

* Point to master till archive is back (#3285)

* Pi-Hole statistics sensor (#3158)

* Add Pi-Hole sensor

* Update docstrings and remove print()

* Use None for payload

* Added stuff for support range setting (#3189)

* cleanup Homematic code (#3291)

* cleanup old code

* cleanup round 2

* remove unwanted platforms

* Update frontend

* Hotfix for #3100 (#3302)

* Fix TP-Link Archer C7 long passwords (#3225)

* Fix tplink C7 long passwords

Fixes an issue where passwords longer than 15 chars could not log in to Archer C7 routers.

* Truncate in correct place

* Add comment about TP-Link C7 pass truncation

* Fix lint error

* Truncate comment at 79 chars not 80

* modbus write registers service (#3252)

* Fix bloomsky platform discovery (#3303)

* Remove dev tag

robbiet480 committed:
Version bump

tchellomello committed:
Fixed voluptuous to accept string instead positive_int for CODE on Simplisafe (#3310)

tchellomello committed:
Fixed voluptuous to accept string instead positive_int for CODE on Simplisafe (#3310)

tchellomello committed:
Fixed voluptuous to accept string instead positive_int for CODE on Simplisafe (#3310)

fabaff committed:
Add timeout (#3304)

fabaff committed:
Migrate to voluptuous (#3298)

fabaff committed:
Migrate to voluptuous (#3290)

fabaff committed:
Migrate to voluptuous (#3292)

fabaff committed:
Use voluptuous for Neurio (#3289)

* Migrate to voluptuous

* Migrate to voluptuous

fabaff committed:
Use constants (#3284)

fabaff committed:
Migrate to voluptuous (#3281)

fabaff committed:
Use constants and update ordering (#3274)

fabaff committed:
Use constants and update ordering (#3275)

fabaff committed:
Migrate to voluptuous (#3282)

fabaff committed:
Migrate to voluptuous (#3280)

fabaff committed:
Migrate to voluptuous (#3277)

fabaff committed:
Migrate to voluptuous (#3276)

fabaff committed:
Extend schema (#3278)

Commit:
Convert rgb to hsb for Wink Osram light

Commit:
Merge pull request #3218 from w1ll1am23/full_color_support_wink_osram

Add full color support for Osram Lightify bulbs with Wink

Teagan42 committed:
Add ISY programs and support for all device types (#3082)

*  ISY Lock, Binary Sensor, Cover devices, Sensors and Fan support
* Support for ISY Programs

fabaff committed:
Use voluptuous for netatmo (#3287)

* Migrate to voluptuous

* Switch back to archive (reverting #3285)

kellerza committed:
Add voluptuous to locative (#3254)

turbokongen committed:
Revert only add 1 device (#3324)

turbokongen committed:
Revert only add 1 device (#3324)

turbokongen committed:
Revert only add 1 device (#3324)

Teagan42 committed:
Automatic Device Tracker Bug Fix (#3330)

* Iterate over items

* Pass display name as host name

Teagan42 committed:
Automatic Device Tracker Bug Fix (#3330)

* Iterate over items

* Pass display name as host name

Teagan42 committed:
Automatic Device Tracker Bug Fix (#3330)

* Iterate over items

* Pass display name as host name

balloob committed:
Comment out pyuserinput in requirements_all (#3307)

* Comment out pyuserinput in requirements_all

* Ignore import error for keyboard component

balloob committed:
Bugfix group order (#3323)

* Add ordered dict config validator

* Have group component use ordered dict config validator

* Improve config_validation testing

* update doc string config_validation.ordered_dict

* validate full dict entries

* Further simplify ordered_dict validator.

* Lint fix

balloob committed:
Bugfix group order (#3323)

* Add ordered dict config validator

* Have group component use ordered dict config validator

* Improve config_validation testing

* update doc string config_validation.ordered_dict

* validate full dict entries

* Further simplify ordered_dict validator.

* Lint fix

balloob committed:
Bugfix group order (#3323)

* Add ordered dict config validator

* Have group component use ordered dict config validator

* Improve config_validation testing

* update doc string config_validation.ordered_dict

* validate full dict entries

* Further simplify ordered_dict validator.

* Lint fix

balloob committed:
Version bump to 0.28.1

balloob committed:
Version bump to 0.28.1

balloob committed:
Merge pull request #3332 from home-assistant/hotfix-0-28-1

Hotfix 0 28 1

clach04 committed:
bluetooth_le_tracker clarify header with LE (#3328)

Khabi committed:
Slack notification optional username / icon (#3314)

* Slack notification optional username / icon

* Small bugfix to handle defaults better

* Dedup'ing code

sam-io committed:
Added link to online docs

sam-io committed:
Added link to online docs

fabaff committed:
Update ordering, constants, and callback name (light.*) (#3339)

* Extend schema

* Update ordering

* Add line breaks

* Align callback name with other platforms

* ALign callbackname with other platforms

* Update callback name to match other platforms

* Update callback name

* Update callback name

fabaff committed:
Fix typo (#3343)

mKeRix committed:
Minor naming fix (#3318)

turbokongen committed:
Bugfix ecobee: inverted high and low temps and enforce int to temps (#3325)

* inverted high and low temps

* Looks like somethings are mixed up

* Added debugentires

* Added debugentires 2

* Enforce int on temperatures

turbokongen committed:
Bugfix ecobee: inverted high and low temps and enforce int to temps (#3325)

* inverted high and low temps

* Looks like somethings are mixed up

* Added debugentires

* Added debugentires 2

* Enforce int on temperatures

turbokongen committed:
Bugfix ecobee: inverted high and low temps and enforce int to temps (#3325)

* inverted high and low temps

* Looks like somethings are mixed up

* Added debugentires

* Added debugentires 2

* Enforce int on temperatures

pvizeli committed:
Update ha-ffmpeg version 0.12 and add tests (#3301)

* update ha-ffmpeg version 0.12 and add tests

* change error logging

pvizeli committed:
Bugfix voluptuous on recorder (#3350)

pvizeli committed:
Bugfix voluptuous on recorder (#3350)

pvizeli committed:
Bugfix voluptuous on recorder (#3350)

turbokongen committed:
Missing garage door detection (#3349)

turbokongen committed:
Missing garage door detection (#3349)

turbokongen committed:
Missing garage door detection (#3349)

DavidLP committed:
Bugfix pilight component (#3355)

* BUG Message data cannot be changed thus use voluptuous to ensure format

* Pilight daemon expects JSON serializable data

Thus dict is needed and not a mapping proxy.

* Add explanation why dict as message data is needed

* Use more obvious voluptuous validation scheme

* Pylint:  Trailing whitespace

DavidLP committed:
Bugfix pilight component (#3355)

* BUG Message data cannot be changed thus use voluptuous to ensure format

* Pilight daemon expects JSON serializable data

Thus dict is needed and not a mapping proxy.

* Add explanation why dict as message data is needed

* Use more obvious voluptuous validation scheme

* Pylint:  Trailing whitespace

DavidLP committed:
Bugfix pilight component (#3355)

* BUG Message data cannot be changed thus use voluptuous to ensure format

* Pilight daemon expects JSON serializable data

Thus dict is needed and not a mapping proxy.

* Add explanation why dict as message data is needed

* Use more obvious voluptuous validation scheme

* Pylint:  Trailing whitespace

Commit:
Add open/closed state for open_cover and close_cover in SERVICE_TO_STATE (#3180)

* Add open/closed state mapping for open_cover and close_cover

* Add 'open', 'closed' for open/close_cover_tilt

* Revert "Add 'open', 'closed' for open/close_cover_tilt"

This reverts commit e45582d4394a33feedfce190a1dba96473d24825.

Commit:
Add open/closed state for open_cover and close_cover in SERVICE_TO_STATE (#3180)

* Add open/closed state mapping for open_cover and close_cover

* Add 'open', 'closed' for open/close_cover_tilt

* Revert "Add 'open', 'closed' for open/close_cover_tilt"

This reverts commit e45582d4394a33feedfce190a1dba96473d24825.

Commit:
Add open/closed state for open_cover and close_cover in SERVICE_TO_STATE (#3180)

* Add open/closed state mapping for open_cover and close_cover

* Add 'open', 'closed' for open/close_cover_tilt

* Revert "Add 'open', 'closed' for open/close_cover_tilt"

This reverts commit e45582d4394a33feedfce190a1dba96473d24825.

balloob committed:
Version bump to 0.28.2

balloob committed:
Version bump to 0.28.2

balloob committed:
Migrate core from threads to async awesomeness (#3248)

* Add event loop to the core

* Add block_till_done to HA core object

* Fix some tests

* Linting core

* Fix statemachine tests

* Core test fixes

* fix block_till_done to wait for loop and queue to empty

* fix test_core for passing, and correct start/stop/block_till_done

* Fix remote tests

* Fix tests: block_till_done

* Fix linting

* Fix more tests

* Fix final linting

* Fix remote test

* remove unnecessary import

* reduce sleep to avoid slowing down the tests excessively

* fix remaining tests to wait for non-threadsafe operations

* Add async_ doc strings for event loop / coroutine info

* Fix command line test to block for the right timeout

* Fix py3.4.2 loop var access

* Fix SERVICE_CALL_LIMIT being in effect for other tests

* Fix lint errors

* Fix lint error with proper placement

* Fix slave start to not start a timer

* Add asyncio compatible listeners.

* Increase min Python version to 3.4.2

* Move async backports to util

* Add backported async tests

* Fix linting

* Simplify Python version check

* Fix lint

* Remove unneeded try/except and queue listener appproriately.

* Fix tuple vs. list unorderable error on version compare.

* Fix version tests

beepmill committed:
Ignore desktop.ini (Windows Explorer) (#3363)

Teagan42 committed:
Automatic polling (#3360)

* Test updating automatic

* Scan interval

* Schedule scan every time delta

* Pass around has

* Recursive issue

* Method invocation

* Oops

* Set up poll

* Default argument value

* Unused import

* Semicolon

* Fix tests

* Linting

* Unneeded throttle as it's handled by time event

* Use track time change event listener

* Disable lint rule

* Attribute removed - removing test

* Debug instead of info

* Unused import

Teagan42 committed:
Automatic polling (#3360)

* Test updating automatic

* Scan interval

* Schedule scan every time delta

* Pass around has

* Recursive issue

* Method invocation

* Oops

* Set up poll

* Default argument value

* Unused import

* Semicolon

* Fix tests

* Linting

* Unneeded throttle as it's handled by time event

* Use track time change event listener

* Disable lint rule

* Attribute removed - removing test

* Debug instead of info

* Unused import

Teagan42 committed:
Automatic polling (#3360)

* Test updating automatic

* Scan interval

* Schedule scan every time delta

* Pass around has

* Recursive issue

* Method invocation

* Oops

* Set up poll

* Default argument value

* Unused import

* Semicolon

* Fix tests

* Linting

* Unneeded throttle as it's handled by time event

* Use track time change event listener

* Disable lint rule

* Attribute removed - removing test

* Debug instead of info

* Unused import

balloob committed:
Merge pull request #3359 from home-assistant/hotfix-0-28-2

0.28.2

fabaff committed:
Use voluptuous for SCSGate (#3265)

* Migrate to voluptuous

* Extend platforms

turbokongen committed:
Fix temp conversion for nest setpoint (#3373)

* Fix temp conversion for nest setpoint

* DEbug

fabaff committed:
Add Simplepush notifications (#3336)

turbokongen committed:
zxt_120 set temperature did not update on setpoint (#3380)

pvizeli committed:
fix xbox live entity id (#3368)

pvizeli committed:
Bugfix auto/manual mode change (#3384)

persandstrom committed:
Modbus component refactoring - sensors and switches (#3297)

ericwclymer committed:
Fix saving push_notification.conf as suggested by @robbiet480 (#3382)

pvizeli committed:
SleepIQ component with sensor and binary sensor platforms (#3390)

Original from #2949

wokar committed:
fixed link constructor to show icon again (#3388)

kellerza committed:
Fix tests (#3387)

daBONDi committed:
Implement Sensor for KNX Platform (#2911)

* Added Sensor Support for KNX Devices

Added Sensor for KNX Group Addresses
- Temperature
- Wind Speed
- Illuminance(LUX)

Mostly to fetch from a KNX Wetterstation

* Some pylint,flake8 fixes

* Pydoc Fixes

* Fix Coverage Ordering

* Refactor KNX Sensor

Refactor to Idea from @usul27 and added Minimum Maximum

* Removed Measurement Untis from const.py

Removed needed Measurement Units from const.py and add it to
sensor\knx.py

* Change .coveragerc

* Add as Requested from @Teagan42 the new Type Names

Additional add CONF_MINIMUM and CONF_MAXIMUM

* Added Changes as Requested from @Teagan42

* Fixed the Merge Conflict, Hopefully i done it rigth :-)

* Fixed Styling

fabaff committed:
Use voluptuous for statsd (#2928)

* Migrate to voluptuous

* Update tests

robbiet480 commented on homeassistant/components/notify/apns.py:
```diff
@@ -0,0 +1,238 @@
+        track_state_change(hass, tracking_ids, state_changed_listener)
+
+    @staticmethod
+    def write_device(out, device):
```
Is this function manually writing out JSON?

sam-io replied:
This writes out YAML, its a [flow mapping](http://www.yaml.org/spec/1.2/spec.html#id2790832) instead of the usual block mapping. It means that if the device name isn't supplied I can have an empty mapping for example:

`123456: {}`

Or if a name is supplied it will look like this:

`123456: { name: Test Phone }`

mKeRix committed:
Added away timeout setting for idle devices (#3321)

robjohnson189 committed:
Vera Thermostat Support (#3247)

* vera thermostat & climate support

* disable abstract

* code review changes

* fix

* fix build

* remove old method

fabaff committed:
Use voluptuous for Yamaha receiver (#3210)

* Migrate to voluptuous

* Add missing configuration variables

nkgilley committed:
short sleep (#3115)

Commit:
Miflora (#3053)

* First version of the MiFlora sensor (not yet finished)

* First workign version

* Added some documentation
Get name from sensor, if not defined

* Ignore IOError

* Added force_update option

* Updated comments

* Renamed fertility to conductivity (what it really is)

* MiFlora library update

* Updated helper files

* Formatting

* Fixed pylint errors

* Removed default from monitored conditions

* Removed KeyError handling as a KeyError should never be raised

* Added a return when no data is received

* emoved unnecessary return statement

* Changed default name

* Changes quotes and string operation ( @Teagan42 )

* - number of samples for median calculation is now configurable
- set state to None if no data could be polled from sensor

* Bugfix in library
more logging

* Fixed miflora version number

fabaff committed:
Upgrade sqlalchemy to 1.0.15 (#3333)

fabaff committed:
Upgrade cherrypy to 8.1.0 (#3334)

fabaff committed:
Upgrade PyMata to 2.13 (#3335)

fabaff committed:
Migrate to voluptuous (#3337)

fabaff committed:
Use voluptuous for iTunes (#3344)

* Migrate to voluptuous

* Add support for SSL/TLS

fabaff committed:
Use voluptuous for KNX (#3345)

* Migrate to voluptuous

* Make host optional and set default

pvizeli committed:
Update yahooweather 0.8 / change request time (#3352) [Breaking change]

jawilson committed:
Fix octoprint sensor (#3385)

* Fix non-temperature sensors for octoprint

* Fix double space in octoprint temperature names

* Fix tox linting errors

fabaff committed:
Use voluptuous for Zone (#3377)

* Migrate to voluptuous

* Zone: Remove unneeded latitude/longitude check

fabaff committed:
Migrate to voluptuous (#3372)

fabaff committed:
Use voluptuous for Tellstick (#3367)

* Migrate to voluptuous

* Update tellstick.py

Commit:
fix style in tellstick sensor

Danielhiversen committed:
Merge pull request #3395 from home-assistant/style_fix

fix style in tellstick sensor

chrom3 committed:
Kodi notification platform (#3403)

ej81 committed:
Bugfix: incorrectly inverted value when setting cover position (#3376)

especially-relative committed:
Add operation_list to radiotherm (#3393)

* Add operation_list to radiotherm

* Use constants

Commit:
Nuimo controller component based on SDK (#3039)

especially-relative committed:
Implemented onkyo reconnect (#3061)

* Implemented onkyo reconnect

Connection object is cleared after a failed command. It will be
automatically recreated upon the next command running. This should allow
for failed connections to be restored.

* Remove reduntant error catching

* Run all update commands with command wrapper.

* Handle errors better

* Removed unused global

I have no idea how that got there.

pvizeli committed:
Split ffmpeg to compoment (#3396)

Add an optional extended description…

pavoni committed:
Add support for Vera covers. (#3411)

balloob committed:
Fix zones (#3413)

jawilson committed:
Add ability to set fan made to Nest climate component (#3399)

* Add ability to set fan made to Nest climate component

* Use constants for fan values

* Use STATE_ON from cost

* Fix lint error

balloob committed:
Use setup_component in tests (#3414)

* Alarm Control Panel Manual - use setup_component

* Update automation - zone tests

* Update climate - demo tests

* Update climate - generic thermostat tests

* Update cover - command line tests

* Update cover - demo tests

* Update device tracker tests

* Update device tracker - owntracks tests

* Update fan - demo tests

* Update garage door - demo tests

* Update light tests

* Update lock - demo tests

* Update media player - demo tests

* Update notify - command line tests

* Update notify - demo tests

* Update notify - file tests

* Update notify - group tests

* Update sensor - mfi tests

* Update sensor - moldindicator tests

* Update sensor - mqtt room tests

* Update switch - command line

* Update switch - flux

* Update switch tests

* Update scene tests

* Fix wrong default port for mfi switch

balloob committed:
Asyncio event helpers (#3415)

* Automation - Event: Use coroutine

* Convert event helpers to coroutine

* Fix linting

* Add hass.async_add_job

* Automation - Event to use async_add_job

balloob committed:
Make track_point_in_utc_time more async (#3428)

* Make track_point_in_utc_time more async

* Make track_point_in_time async friendly

lwis committed:
Update pychromecast (#3416)

pvizeli committed:
Bugfix if ffmpeg is not present on config / Update ha-ffmpeg 0.13 (#3418)

Danielhiversen committed:
Fix issue #3401 weblink (#3402)

fabaff committed:
Use voluptuous for logger (#3375)

* Migrate to voluptuous

* No list for configuration check

fabaff committed:
Use voluptuous for RPi GPIO (#3371)

* Migrate to voluptuous

* Remove the check for lists

deisi committed:
Recieve signals from a keyboard and use keyboard as a remote control (#3305)

balloob committed:
Update Docker to use Python 3.5 (#3430)

joyrider3774 committed:
Emoncms feeds sensor component (#3258)

* Add Initial version for emoncms feeds sensor

* flake8 test fixes

* pylint test fixes

* - fix bug with include_feed_id_names not assigning the name to the  element in the same postion as found in include_feed_id
- a few structure changes to have less nesting (pylint fix)
- minor other changes

* update .coveragerc

* voluptuous fixes:
- exclude_feed_id and include_feed_id Exclusive group so that only one (or none) can be specified at once
- id must be positive int
- exclude_feed_id and include_feed_id must be positive int

* Fix comment so it refers to the documentation

* use string formatting

* Remove outer try

* clean up sensors.append calls
(break them out for loop)

* multiple changes like:
- rename config value "include_feed_id" to "include_only_feed_id"
- rename config value "include_feed_id_names" to "sensor_names"
- renamed config value sensor_names is now a dictionary an can also be used when renamed config value "include_only_feed_id" is not specified
- Set default value for scan_interval using the config validation
- blank lines between default, 3rd party and own imports
- removed homeassistant.util import, it was not needed anymore

* fix extended voluptuous schema
scan_interval should not be extended to PLATFORM_SCHEMA as it was already in the schema by config_validation helper so i should not add / change it. It was also causing problems reading the value from the config.

* Use Home Assistant polling

* remove statement that can never happen

* Guard clause

* Reduce instance variables

kellerza committed:
Use voluptuous for nmap, tplink, thomson device trackers (#3124)

* Use Voluptuous for nmap, tplink, thomson device_trackers

* Fix logic

philhawthorne committed:
Add PyBluez to Dockerfile (#3423)

* Add PyBluez to Dockerfile

Adds PyBluez to Dockerfile so people using Docker can run Bluetooth
devices

* Remove pip install of pybluez

Pybluez will be installed automatically when the Bluetooth device
tracker is enabled

balloob committed:
Update frontend

pvizeli committed:
Add toggle service to input_boolean (#3432)

mezz64 committed:
Update pyenvisalink to latest version (#3435)

Danielhiversen committed:
Accept login from approved ips without password (#3427)

technicalpickles committed:
Update Ecobee state after making changes to climate (#3436)

* Update Ecobee state after making changes to climate

Without this, climate and sensor state will take up to 3 minutes
(the MIN_TIME_BETWEEN_UPDATES on its update throttle) to update in
the interface, which makes it more difficult to do automation around the
state.

* Use a boolean instance variable that update can check, rather than always calling update

balloob committed:
Migrate camera.uvc to voluptuous (#3440)

balloob committed:
Feature/voluptuous influxdb (#3441)

* Migrate to voluptuous

* Fix voluptuous influxdb

balloob committed:
Better Hue error reporting (#3443)

bbangert committed:
Fix test shutdown to ensure loop/threads are clean. (#3447)

* Fix test shutdown to ensure loop/threads are clean.

We now ensure the loop is closed, it has completed, and the
executer has completed. This ensure all threads are freed
up with any test calling hass.stop().

* Fix lint issue with run_loop

balloob committed:
Fix slow tests (#3444)

* Fix RFXtrx tests

* Report slow tests on CI

* Minor rfxtrx clean up

* rfxtrx test tweak

balloob committed:
Remove validate_config (#3448)

Teagan42 committed:
Isy fixes (#3457)

* Fix binary sensor

* Add 'stopped' to states

* Add '%' to states for light

* ISY light brightness support

* Method case

* Z-Wave unit 51 is a light

balloob committed:
Update frontend

balloob committed:
Fix logger config validation (#3459)

fabaff committed:
Add unit to comment (#3452)

tchellomello committed:
Added support to Yahoo Finance to track the stock market within Home Assistant (#3446)

* Added support to Yahoo Finance to track the stock market within Home Assistant

* Fixed pylint issues

* Fixed formatting issues

* Fixed pep257 issues

*   - Fixed URL link
  - Added attributes for Yahoo Finance

* Removed price sales ATTR

* Fixed lint and flake8 issues. Added attribution to Yahoo! per https://developer.yahoo.com/attribution/

milaq committed:
Allow hiding automation entities from UIs (#3442)

* Allow hiding automation entities from UIs

* Flake8 fixes: Allow hiding automation entities from UIs

* Automation: Rework hide entity feature

 * Refactor keyword 'hidden' to 'hide_entity' to avoid ambiguity
 * Migrate hide_entity subsetting to Voluptuous

balloob committed:
Move worker pool monitoring to be time based instead of add_job based. (#3439)

* Move worker pool monitoring to be time based instead of add_job based.

* Stub out worker pool monitor during tests

* Add test for monitor worker pool.

* Improve naming

* Test stop_monitor coroutine

* Add async_create_timer test

* Finish rename create_timer

balloob committed:
Update frontend

Commit:
Sensor updates (#3410)

lwis committed:
Use uvloop for asyncio policy (#3417)

sam-io committed:
Email (#3421)

* Added email component

* added email sensor component

* added doc string to test class

* fixed lint error

* fixed lint error

* rename of email component

* added another block as test fails on CI

* added retry to multi email test

* added delay to retry

* added to .coveragerc

* removed sleep from tests and fixed up stale comments

LinuxChristian committed:
D-link switch library bump and error handling for W110 devices (#3386) (#3425)

sam-io commented:
Anything else you want me to do with this to get it merged?

fabaff committed:
Use voluptuous for Hue (#3340)

* Migrate to voluptuous

* Change name used in test

milas committed:
Allow pairing with Harmony Hub (#3467)

The real Hue hub responds to both `/api` and `/api/`. For greater
compatibility, the view now responds to both using `extra_urls`.

wokar committed:
Filter entities from logbook (#3426)

* o added ability to exclude entities or domains from logbook
o exclude hidden entities

* fixed remaned configuration key

* - filter the events before they get passed to humanify, to separate concerns
- instead of looking at customize, look for the hidden attribute on the state change events
- access to configuration defaults to an empty list - no need to check

* - filter only events of type EVENT_STATE_CHANGED
- improve config handling

* added unit tests to cover all filter cases and logbook message creation

balloob commented:
@robbiet480

Commit:
Adding support for a white value  (#3338)

* Update __init__.py

addedattribute "WHITE_VALUE" to improve support for RGBW devices

* Update services.yaml

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

* Update __init__.py

shortened line

* Update __init__.py

* Update __init__.py

* Add mysensors RGBW and light tests

* Activate support for mysensors RGBW devices with support for
	white_value attribute.
* Add white_value support in light demo platform.
* Add tests for white_value and more for light component.
* Add tests for light demo platform.
* Fix import order in check_config.

lwis committed:
Add additional fields to influx (#3462)

fabaff committed:
Migrate to voluptuous (#3342) [Breaking Change]

simonszu committed:
Add the ability to manually specify a Yamaha AVR via it's IP address (#3451)

* Added the possibility to manually specify a Yamaha Receiver

* Added the possibility to manually specify a Yamaha AVR

* Using string formatting

* Hostname checks for None now

* Do not use add_devices for each if-branch separately

* Fixed linting

jabesq committed:
Add specific icon for forecast.io sensors (#3465)

Cinntax committed:
Envisalink Fixes + Enhancements (#3460)

* Added the ability to trigger the alarm.

* Bump version of pyenvisalink to 1.3

* Fixed an issue where the panic_type was not passed to the sub-components properly.

* Bump pyenvisalink version, and make default panic mode = police.

* Pass in event loop to pyenvisalink.

* Made the components play nicely with asyncio.

* Bump pyenvisalink to 1.6

* Bump up pyenvisalink, and better handle synchronous setup.

Teagan42 committed:
UOM is a list - not a string. (#3469)

balloob committed:
Merge branch 'master' into potential-master-fix

balloob committed:
Fix missing commits from 0.28

Add missing commits to master

tinglis1 committed:
Bom weather current component (#3370)

* add bom_weather_current component

* wrong fork

* wrong fork

* Code tidy up

- fixed order of imports
- changes some of the units to be standardised

* Rename bom_weather_current.py to bom.py

* pylint changes

* lint formatting changes

* formatting

* fix logger string variable formatting

* Update .coveragerc

pvizeli committed:
convert first to string befor matching (#3476)

balloob committed:
Fix nmap config (#3482)

kaustubhphatak committed:
Add support for mysensors HVAC device (#3405)

* Added Support for mysensnors Climate/HVAC device

* Added Support for mysensnors-hvac device:fix pylint error

* Added Support for mysensnors-hvac device:fix pylint error2

* Fixed Issues in code as per review comments

* Fixed Linter Errors

* Fixed Linter Errors:2

* Fixed Linter Errors:2

* Fixed Linter Errors

* Fixed Linter Errors

* Fixed Linter Errors

* Added Support for MySensors HVAC| Fixed Review Comments| Removed Additional Comments

* Added Support for MySensors HVAC| Fixed Review Comments Itr2

* Changes to correctly support devices with both high and low bound temperatures

* Changed to optimize the code

irvingwa committed:
Added check for channel in kodi media player to play channel from PVR. (#3475)

* Added check for channel in kodi media player to play channel from PVR.

* test

Commit:
Update rfxtrx to 0.12

sytone committed:
Added play media to squeezebox (#3306)

* Added play media to squeezebox

The squeezebox component can now add a URI to an existing playlist or just over write it to force a stream to play.

* Cleaned up flake8 issues with formatting. 

Spacing... The end of the world! Fixed. Once day the tools will fix this on the fly, one day...

[x] ./homeassistant/components/media_player/squeezebox.py:307:1: W293 blank line contains whitespace
[x] ./homeassistant/components/media_player/squeezebox.py:366:1: W391 blank line at end of file
[x] ./homeassistant/components/media_player/squeezebox.py:366:1: W293 blank line contains whitespace

Updated SUPPORT_SQUEEZEBOX to add SUPPORT_PLAY_MEDIA

[x] ./homeassistant/components/media_player/squeezebox.py:13:1: F401 'homeassistant.components.media_player.SUPPORT_PLAY_MEDIA' imported but unused

* Updates from review

Updated the comments to indicate they are developer / API comments and not for end users.
Marked the private functions with a leading underscore (_)

* Fixed Lint issues. 

202ERROR: InvocationError: '/home/travis/build/home-assistant/home-assistant/.tox/lint/bin/flake8'

203lint runtests: commands[1] | pylint homeassistant

204************* Module homeassistant.components.media_player.squeezebox

205C:322, 0: Trailing whitespace (trailing-whitespace)

Commit:
Added more configuration parameters (#3479)

Upgraded miflora library to 0.1.9 (which is more stable)

kellerza committed:
Sorted yaml output for check_config (#3354)

* Consistent display of check_config dicts

* OrderedDict

* remove sorted

kellerza committed:
Use voluptuous for input_slider, input_boolean, input_select (#3256)

* Use voluptuous for input slider

* floats

* _setup_component

* Imperative mood

* CONFIG_SCHEMA

* None returns empty ensure_list

* allow_extra

* bool

* restore ensure_list behaviour

Danielhiversen committed:
Merge pull request #3489 from home-assistant/rfxtrx_update

Update rfxtrx lib to 0.12

fabaff committed:
Update links (#3488)

fabaff committed:
Fix typos (#3490)

fabaff committed:
Fix PEP257 issues (#3491)

* Align test name with platform

* Fix PEP257 issues

fabaff committed:
Fix lint issues (#3492)

MartinHjelmare committed:
Fix mysensors required version for HVAC (#3499)

kellerza committed:
Add domain to boolean (#3500)

robbiet480 committed:
Fix all_covers group friendly_name

It now matches other components (`all covers` instead of `all_covers`).

robbiet480 committed:
Fix ENTITY_ID_ALL_COVERS format

kk7ds committed:
Bump somecomfort to 0.3.2 (#3502)

This has fixes related to #3468

balloob committed:
Filter out falsey platform configs

balloob committed:
Fix zone being setup twice

pvizeli committed:
Add new component for licence plates processing (OpenAlpr) (#3461)

* Add new component for licence plates processing (OpenAlpr)

* address balloobbot comments

* add to coveragerc

* move config from device to base

* fix lint

* move local api test to voluptous

* split render engine

* change cloud_api pip string & lint

* update requirements_all.txt

* fix lint

* update cloud_api url

* convert base64 byte string to string

* Update cloudapi / add configence / add state

* fix lint

* change state to high confidence plate

* fix cloudapi

* fix local api detection

* add wraper for local api

* fix lint

* fix wrong import

* fix HAAlpr name

* update ha-alpr without async

* support only eventbased requests with interval 0

* fix minor things

* fix lint

* fix lint2

balloob committed:
Merge pull request #3503 from home-assistant/fix-platform-component-no-config

Allow platform components without config

balloob committed:
Merge branch 'master' into dev

Conflicts:
	homeassistant/components/climate/ecobee.py
	homeassistant/components/cover/wink.py
	homeassistant/const.py

MartinHjelmare committed:
Fix mysensors white value (#3508)

* Fix turning on mysensors light with white value attribute in kwargs.
* Fix import order in check_config.py.

robbiet480 committed:
Change notify target property to be a dictionary (#3501)

* Change notify target property to be a dictionary

* Make demo target properties unique and fix test to match behavior

balloob committed:
Pre-compile templates (#3515)

* Pre-compile templates

* Compile templates in numeric_state condition

pvizeli committed:
Use setup_component in tests v1 (#3507)

* update unittests like #3414

* setup_component - splunk

* setup_component - statsd

* fix statsd & splunk unittest config values

* component_setup - device_sun_light_trigger

* setup_component - introduction

* component_setup - persistent_notification

* setup_component - logentries, mqtt eventstream

* fix unittest logentries

MartinHjelmare committed:
Fix mqtt cover retain and state (#3519)

* Platform schema had duplicate retain keys, which made it always set
	to default.
* Optimistic state changed was inverted, due to using integer position
	instead	of boolean.
* Add more tests for mqtt cover.

pvizeli committed:
Use setup_component in tests v2 (#3537)

* setup_component - sun

* setup_component - updater

* setup_component - weblink

Cinntax committed:
Added a dispatch call to the envisalink sensor to also get partition status updates.

fabaff committed:
Change line separator to lf instead of crlf (#3533)

* Change line separator to lf instead of crlf

* Update permissions

* Use LN instead of CRLF

* Remove BOM and use LN instead of CRLF

persandstrom committed:
modbus update, add error log instead of AttributeError exception (#3517)

Commit:
update rfxtrx lib

Danielhiversen committed:
Merge pull request #3542 from home-assistant/rfxtrx_lib

update rfxtrx lib

turbokongen committed:
Update Climate slider code (#3394)

* Update ecobee to use only range setpoints

* Update nest to only use range setpoints

* Update demo to use only range for ecobee test device

* Update test

* Fetch unit from ecobee

* generic_thermostat did not have state

* generic_thermostat test update

balloob committed:
Optimize template 2 (#3521)

* Enforce compiling templates

* Refactor templates

* Add template validator to Logbook service

* Some more fixes

* Lint

* Allow easy skipping of rfxtrx tests

* Fix template bug in AND & OR conditions

* add entities extractor

Conflicts:
	tests/helpers/test_template.py

* fix unittest

* Convert template to be async

* Fix Farcy

* Lint fix

* Limit template updates to related entities

* Make template automation async

fabaff committed:
Add link to error message (#3530)

balloob committed:
Limit worker pool to 10 threads (#3560)

* Limit worker pool to 10 threads

* Comment evdev in requirements

* Allow skipping RFXtrx tests locally

* Fix worker pool size tests

* lol whut

fabaff committed:
Upgrade dnspython3 to 1.14.0 (#3522)

fabaff committed:
Upgrade fuzzywuzzy to 0.12.0 (#3524)

fabaff committed:
Upgrade python-telegram-bot to 5.1.0 (#3525)

balloob committed:
Version bump to 0.29

balloob committed:
Version bump to 0.30.0.dev0

tchellomello committed:
Fixed typo (#3569)

Sep 29 00:59:22 pi hass[21333]: if self.device.measurment_scale == 'F':
Sep 29 00:59:22 pi hass[21333]: AttributeError: 'Device' object has no attribute 'measurment_scale'

pavoni committed:
Bump library to catch more exceptions in poll thread.

pavoni committed:
Merge pull request #3577 from home-assistant/fix_loopenergy_exceptions

Loopenergy library - bump to catch more exceptions in poll thread.

fabaff committed:
Remove duplicate port entry (fixes #3583) (#3584)

jabesq committed:
Netatmo: Hotfix for hass 0.29 (#3576)

Signed-off-by: Hugo D. (jabesq) <jabesq@gmail.com>

pvizeli committed:
Bugfix voluptuous for hue (#3589)

sam-io committed:
Merge branch 'dev' of github.com:sam-io/home-assistant into dev

bbangert commented:
The new HTTP 2.0 based APNS should be used, the prior version (using the apns library) has some drawbacks for reliability. It's also quite a bit simpler, here's one we made for our push service: https://github.com/mozilla-services/autopush/blob/master/autopush/router/apns2.py

sam-io committed:
changed to use http/2 library for push notifications

sam-io committed:
fixed lint issue

sam-io committed:
fixed test that fails on CI

sam-io committed:
another go at fixing test that fails on CI

sam-io committed:
another go at fixing test that fails on CI

sam-io committed:
another go at fixing test that fails on CI

sam-io commented:
@bbangert I've changed it to use a library that uses the new http/2 api

robbiet480 commented:
@sam-io I'm unsure how so many commits got into this PR. Can you please open a new PR against the latest version of master with your changes? Thanks!
